internal/service/certificatesigningrequest.go

```diff
@@ -56,9 +57,16 @@ func signApprovedCertificateSigningRequest(ca *crypto.CA, request api.Certificat
 
 // (DELETE /api/v1/certificatesigningrequests)
 func (h *ServiceHandler) DeleteCertificateSigningRequests(ctx context.Context, request server.DeleteCertificateSigningRequestsRequestObject) (server.DeleteCertificateSigningRequestsResponseObject, error) {
+	allowed, err := auth.GetAuthZ().CheckPermission(ctx, "certificatessigningrequests", "deletecollection")
+	if err != nil {
+		return server.DeleteCertificateSigningRequests401JSONResponse{Message: fmt.Sprintf("auth failed: %v", err)}, nil
+	}
+	if !allowed {
+		return server.DeleteCertificateSigningRequests403JSONResponse{Message: "cannot delete certificate signing requests"}, nil
```

Comment:
Could we define an error constant for this for consistency? I think a "Forbidden" would even be enough, as the rest is clear from the API call itself.